test/decoders/VideoDecoderTest.cpp

```diff
@@ -223,6 +223,28 @@ TEST_P(VideoDecoderTest, DecodesFramesInABatchInNCHW) {
   EXPECT_TRUE(torch::equal(tensor[1], tensorTime6FromFFMPEG));
 }
 
+void testDecoderWithColorConversion(const std::string& colorConversionLibrary) {
+  std::string path = getResourcePath("nasa_13013.mp4");
+  auto preAllocatedOutputTensor = torch::empty({270, 480, 3}, {torch::kUInt8});
+
+  std::unique_ptr<VideoDecoder> ourDecoder =
+      createDecoderFromPath(path, GetParam());
```

Comment:
So compilation fails on this line because I assume `GetParam` is only valid within a `TEST_P` function.

@ahmadsharif1 @scotts any tip on how to write this test over both `swscale` and `filtergraph` without just duplicating the code?

Reply:
gtest has parameterized tests:

https://google.github.io/googletest/reference/testing.html#TEST_P

Reply:
You could call getparam from the parent and pass it down here?

The other way of doing this is parameterized classes but that seems overkill to me.

Reply:
Thanks for your feedback. I tried a few things, but couldn't get it to work (new kind of error each time). I just went for duplicating the whole thing, I hope that's OK. We already have some duplicated tests in this class.